test/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner_test.rb

```diff
@@ -195,4 +202,71 @@ def self.it_triggers_compilation_process
       end
     end
   end
+
+  describe ".check_system_dependencies!" do
+    subject { as_task_runner.check_system_dependencies! }
+
+    describe "when npm is not installed" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns([nil, mock(success?: false)])
+        assert_raises Script::Layers::Infrastructure::Errors::NoDependencyInstalledError do
+          subject
+        end
+      end
+    end
+
+    describe "when node is not installed" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns(["5.2.0", mock(success?: true)])
+        ctx.expects(:capture2e)
+          .with("node", "--version")
+          .returns([nil, mock(success?: false)])
+        assert_raises Script::Layers::Infrastructure::Errors::NoDependencyInstalledError do
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is below minimum" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns([nil, mock(success?: false)])
+
+        assert_raises Script::Layers::Infrastructure::Errors::NoDependencyInstalledError do
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is above minimum and node is below minimum" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns(["5.2.0", mock(success?: true)])
+        ctx.expects(:capture2e)
+          .with("node", "--version")
+          .returns(["v14.4.0", mock(success?: true)])
+        assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+          subject
+        end
+      end
+    end
+
+    describe "when npm version and node are above minimum" do
+      it "should install successfully" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns(["5.2.0", mock(success?: true)])
+        ctx.expects(:capture2e)
+          .with("node", "--version")
+          .returns(["v14.5.0", mock(success?: true)])
+        subject
+      end
+    end
+  end
```

Comment:
In this structure, there are some cases that wouldn't execute. For instance, if `npm version is below minimum` it would raise the error, but it never execute any of the node version checks.